eng/cake/dotnet.cake

```diff
@@ -205,6 +208,42 @@ Task("dotnet-pack")
         Zip(assetsDir, $"{assetsDir}.zip");
     });
 
+Task("dotnet-pack-library-packs")
+    .Does(() =>
+    {
+        var tempDir = $"./artifacts/library-packs-temp";
+
+        var destDir = $"./artifacts/library-packs";
+        EnsureDirectoryExists(destDir);
+        CleanDirectories(destDir);
+
+        void Download(string id, string version, params string[] sources)
+        {
+            version = XmlPeek("./eng/Versions.props", "/Project/PropertyGroup/" + version);
+
+            NuGetInstall(id, new NuGetInstallSettings
+            {
+                Version = version,
+                ExcludeVersion  = true,
+                OutputDirectory = tempDir,
+                Source = sources,
+            });
+
+            CopyFiles($"{tempDir}/**/" + id + ".nupkg", destDir, false);
+            CleanDirectories(tempDir);
+        }
+
+        Download("Microsoft.Maui.Graphics", "MicrosoftMauiGraphicsVersion", "https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6/nuget/v3/index.json");
+        Download("Microsoft.Maui.Graphics.Win2D.WinUI.Desktop", "MicrosoftMauiGraphicsVersion", "https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6/nuget/v3/index.json", "https://api.nuget.org/v3/index.json");
+        Download("Microsoft.WindowsAppSDK", "MicrosoftWindowsAppSDKPackageVersion", "https://api.nuget.org/v3/index.json");
+        Download("Microsoft.Windows.SDK.BuildTools", "MicrosoftWindowsSDKBuildToolsPackageVersion", "https://api.nuget.org/v3/index.json");
```

Comment:
It looks like the msi generation tooling is expecting these files to have the package version in the file name:
```
Workload pack package does not exist: D:\a\_work\1\a\nugets\signed\Microsoft.Maui.Graphics.6.0.300-rc.1.1198.nupkg
Workload pack package does not exist: D:\a\_work\1\a\nugets\signed\Microsoft.Maui.Graphics.Win2D.WinUI.Desktop.6.0.300-rc.1.1198.nupkg
Workload pack package does not exist: D:\a\_work\1\a\nugets\signed\Microsoft.WindowsAppSDK.1.0.1.nupkg
Workload pack package does not exist: D:\a\_work\1\a\nugets\signed\Microsoft.Windows.SDK.BuildTools.10.0.22000.194.nupkg
```

We'll need to add the version to file name on download or when copying the file.

Reply:
Fixed this and the msi seem to be in the artifacts.